pypicloud/access/aws_secrets_manager.py

```diff
@@ -0,0 +1,181 @@
+        kwargs = super(AWSSecretsManagerAccessBackend, cls).configure(settings)
+        kwargs['settings'] = settings
+        kwargs['region'] = settings['auth.region']
+        kwargs['secret_id'] = settings['auth.secret_id']
+        credentials = {}
+
+        if settings.get('auth.access_key'):
+            credentials['aws_access_key_id'] = settings['auth.access_key']
+        if settings.get('auth.secret_key'):
+            credentials['aws_secret_access_key'] = settings['auth.secret_key']
+        if settings.get('auth.session_token'):
+            credentials['aws_session_token'] = settings['auth.session_token']
+        if settings.get('auth.profile_name'):
+            credentials['profile_name'] = settings['auth.profile_name']
+        kwargs['credentials'] = credentials
+        return kwargs
+
+    def _fetch_credentials(self):
+        """ Hit a server endpoint and return the json response """
+        session = boto3.session.Session(**self.credentials)
+        client = session.client(
+            service_name='secretsmanager',
+            region_name=self.region,
+            endpoint_url="https://secretsmanager.{}.amazonaws.com".format(
+                self.region
+            )
+        )
+
+        try:
+            return json.loads(client.get_secret_value(
+                SecretId=self.secret_id
+            )['SecretString'])
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                raise Exception(
+                    "The requested secret " + self.secret_id +
+                    " was not found")
+            elif e.response['Error']['Code'] == 'InvalidRequestException':
+                raise Exception("The request was invalid due to:", e)
+            elif e.response['Error']['Code'] == 'InvalidParameterException':
+                raise Exception("The request had invalid params:", e)
+
+    def verify_user(self, username, password):
+        credentials = self._fetch_credentials()
+        try:
+            return credentials.get('users', {})[username] == password
+        except KeyError:
+            return False
+
+    def _get_password_hash(self, username):
+        # We don't have to do anything here because we overrode 'verify_user'
+        pass
+
+    def groups(self, username=None):
+        credentials = self._fetch_credentials()
+        if not username:
+            return list(credentials.get('groups', {}).keys())
+        groups = []
+        for group_name, users in credentials.get('groups', {}).items():
+            if username in users:
+                groups.append(group_name)
+        return groups
+
+    def group_members(self, group):
+        credentials = self._fetch_credentials()
+        return list(credentials.get('groups', {}).get(group, []))
+
+    def is_admin(self, username):
+        credentials = self._fetch_credentials()
+        admins = credentials.get('groups', {}).get('admins', [])
```

Comment:
I would recommend putting `admins` at the top level instead of nesting it under `groups`, since it serves a slightly different purpose.

Reply:
agreed